src/subdomains/generic/user/services/webhook/webhook.service.ts

```diff
@@ -1,103 +1,88 @@
-    await this.triggerUserDataWebhook(userData, WebhookDataMapper.mapKycData(userData), WebhookType.KYC_CHANGED);
+    for (const user of userData.users) {
+      await this.create({
+        data: JSON.stringify(WebhookDataMapper.mapKycData(userData)),
+        type: WebhookType.KYC_CHANGED,
+        user,
+      });
+    }
   }
 
   async kycFailed(userData: UserData, reason: string): Promise<void> {
-    await this.triggerUserDataWebhook(userData, WebhookDataMapper.mapKycData(userData), WebhookType.KYC_FAILED, reason);
+    for (const user of userData.users) {
+      await this.create({
+        data: JSON.stringify(WebhookDataMapper.mapKycData(userData)),
+        reason,
+        type: WebhookType.KYC_FAILED,
+        user,
+      });
+    }
   }
 
   async fiatCryptoUpdate(user: User, payment: BuyCryptoExtended): Promise<void> {
-    await this.triggerUserWebhook(user, WebhookDataMapper.mapFiatCryptoData(payment), WebhookType.PAYMENT);
+    await this.create({
+      user,
+      data: JSON.stringify(WebhookDataMapper.mapFiatCryptoData(payment)),
+      type: WebhookType.PAYMENT,
+    });
   }
 
   async cryptoCryptoUpdate(user: User, payment: BuyCryptoExtended): Promise<void> {
-    await this.triggerUserWebhook(user, WebhookDataMapper.mapCryptoCryptoData(payment), WebhookType.PAYMENT);
+    await this.create({
+      user,
+      data: JSON.stringify(WebhookDataMapper.mapCryptoCryptoData(payment)),
+      type: WebhookType.PAYMENT,
+    });
   }
 
   async cryptoFiatUpdate(user: User, payment: BuyFiatExtended): Promise<void> {
-    await this.triggerUserWebhook(user, WebhookDataMapper.mapCryptoFiatData(payment), WebhookType.PAYMENT);
+    await this.create({
+      user,
+      data: JSON.stringify(WebhookDataMapper.mapCryptoFiatData(payment)),
+      type: WebhookType.PAYMENT,
+    });
   }
 
   async fiatFiatUpdate(user: User, payment: BuyFiatExtended): Promise<void> {
-    await this.triggerUserWebhook(user, WebhookDataMapper.mapFiatFiatData(payment), WebhookType.PAYMENT);
+    await this.create({
+      user,
+      data: JSON.stringify(WebhookDataMapper.mapFiatFiatData(payment)),
+      type: WebhookType.PAYMENT,
+    });
   }
 
   // --- HELPER METHODS --- //
 
-  private async triggerUserDataWebhook<T extends PaymentWebhookData | KycWebhookData>(
-    userData: UserData,
-    data: T,
-    type: WebhookType,
-    reason?: string,
-  ): Promise<void> {
-    userData.users = await this.userRepo.find({
-      where: { userData: { id: userData.id } },
-      relations: ['wallet', 'userData'],
+  private async create(dto: CreateWebhookInput): Promise<Webhook> {
+    const existing = await this.webhookRepo.findOne({
+      where: {
+        data: dto.data,
+        type: dto.type,
+        reason: dto.reason,
+        user: { id: dto.user.id },
+        sentDate: IsNull(),
+      },
+      relations: { user: true },
     });
+    if (existing) throw new BadRequestException('Webhook already created');
```

Comment:
Do we need to check, if the wallet of the user has a webhook URL configured? Doesn't make sense to store the webhook otherwise ... WDYT?